bpf/lib/common.h

```diff
@@ -476,6 +476,35 @@ struct lb6_service {
 	__u16 weight;
 } __attribute__((packed));
 
+struct lb6_key_v2 {
+	union v6addr address;	/* Service virtual IPv6 address */
+	__be16 dport;		/* L4 port filter, if unset, all ports apply */
+	__u16 slave;		/* Backend iterator, 0 indicates the master service */
+	__u8 proto;		/* L4 protocol, currently not used (set to 0) */
+	__u8 pad;
```

Comment:
My bad - I've been using the `packed` attribute (removed), and I've never checked what layout the compiler generates w/o the attribute being set.

Anyway, after the change we have the following layouts:
```
struct lb4_key_v2 {
        uint32_t                   address;              /*     0     4 */
        uint16_t                   dport;                /*     4     2 */
        uint16_t                   slave;                /*     6     2 */
        uint8_t                    proto;                /*     8     1 */
        uint8_t                    pad[3];               /*     9     3 */

        /* size: 12, cachelines: 1, members: 5 */
        /* last cacheline: 12 bytes */
};
struct lb6_key_v2 {
        union v6addr               address;              /*     0    16 */
        uint16_t                   dport;                /*    16     2 */
        uint16_t                   slave;                /*    18     2 */
        uint8_t                    proto;                /*    20     1 */
        uint8_t                    pad[3];               /*    21     3 */

        /* size: 24, cachelines: 1, members: 5 */
        /* last cacheline: 24 bytes */
};
```

BTW, is there any requirement for alignment (x86_64 doesn't enforce one)?

Reply:
I *suspect* this was previously more about our attempts to match alignment between the C side here and Golang structures, rather than any other alignment requirement. (/cc @borkmann )